src/test/java/io/vertx/ext/asyncsql/SQLTestBase.java

```diff
@@ -658,14 +661,50 @@ protected void checkConsistency(TestContext context, Async async, SQLConnection
 
   private void setupSimpleTable(SQLConnection conn, Handler<AsyncResult<Void>> handler) {
     conn.execute("BEGIN",
-        ar -> conn.execute("DROP TABLE IF EXISTS test_table",
-            ar2 -> conn.execute(CREATE_TABLE_STATEMENT,
-                ar3 -> conn.update("INSERT INTO test_table (id, name) VALUES " + Data.get(),
-                    ar4 -> conn.execute("COMMIT", handler::handle)))));
+      ar -> conn.execute("DROP TABLE IF EXISTS test_table",
+        ar2 -> conn.execute(CREATE_TABLE_STATEMENT,
+          ar3 -> conn.update("INSERT INTO test_table (id, name) VALUES " + Data.get(),
+            ar4 -> conn.execute("COMMIT", handler)))));
   }
 
 
   private static final String CREATE_TABLE_STATEMENT = "CREATE TABLE test_table " +
-      "(id BIGINT, name VARCHAR(255))";
+    "(id BIGINT, name VARCHAR(255))";
+
+  @Test
+  public void testSimpleStream(TestContext context) {
+    Async async = context.async();
+    client.getConnection(ar -> {
+      if (ar.failed()) {
+        context.fail(ar.cause());
+        return;
+      }
+
+      // Create table
+      conn = ar.result();
+      setupSimpleTable(conn, ar2 -> {
 
+        conn.queryStream("SELECT name FROM test_table", ar3 -> {
+          if (ar3.failed()) {
+            context.fail(ar3.cause());
+          } else {
+            final SQLRowStream res = ar3.result();
+            context.assertNotNull(res);
+
+            final AtomicInteger cnt = new AtomicInteger();
+
+            res
+              .handler(row -> {
+                context.assertTrue(Data.NAMES.contains(row.getString(0)));
```

Comment:
If you use `ORDER BY name` in the SQL statement, it should be possible to use `cnt` to check that the result is correct.

I'd also prefer `count` or something more explicit instead of `cnt` but that's not a blocker to me.